hybrasyl/CreatureStatus.cs

```diff
@@ -0,0 +1,265 @@
+                    User?.Say(effect.Messages.Shout);
+            }
+        }
+
+
+        private void ProcessConditions(ModifierEffect effect)
+        {
+            if (effect.Conditions.Set != 0)
+                Target.Condition.Conditions|= effect.Conditions.Set;
+            if (effect.Conditions.Unset != 0)
+                Target.Condition.Conditions &= ~effect.Conditions.Unset;
+        }
+
+        private void ProcessStatModifiers(Statuses.StatModifiers effect, bool remove = false)
+        {
+            if (effect == null) return;
+
+            if (remove)
+            {
+                Target.BonusStr -= effect.Str;
+                Target.BonusInt -= effect.Int;
+                Target.BonusWis -= effect.Wis;
+                Target.BonusCon -= effect.Con;
+                Target.BonusDex -= effect.Dex;
+                Target.BonusHp -= effect.Hp;
+                Target.BonusMp -= effect.Mp;
+                Target.BonusHit -= effect.Hit;
+                Target.BonusDmg -= effect.Dmg;
+                Target.BonusAc -= effect.Ac;
+                Target.BonusRegen -= effect.Regen;
+                Target.BonusMr -= effect.Mr;
+                Target.BonusDamageModifier = effect.DamageModifier;
+                Target.BonusHealModifier = effect.HealModifier;
+                Target.BonusReflectChance -= effect.ReflectChance;
+                Target.BonusReflectIntensity -= effect.ReflectIntensity;
+                if (effect.OffensiveElement == (Statuses.Element) Target.OffensiveElementOverride)
+                    Target.OffensiveElementOverride = Enums.Element.None;
+                if (effect.DefensiveElement == (Statuses.Element)Target.DefensiveElementOverride)
+                    Target.DefensiveElementOverride = Enums.Element.None;
+                Target.BonusAc -= effect.Str;
+            }
+            else
+            {
+                Target.BonusStr += effect.Str;
+                Target.BonusInt += effect.Int;
+                Target.BonusWis += effect.Wis;
+                Target.BonusCon += effect.Con;
+                Target.BonusDex += effect.Dex;
+                Target.BonusHp += effect.Hp;
+                Target.BonusMp += effect.Mp;
+                Target.BonusHit += effect.Hit;
+                Target.BonusDmg += effect.Dmg;
+                Target.BonusAc += effect.Ac;
+                Target.BonusRegen += effect.Regen;
+                Target.BonusMr += effect.Mr;
+                Target.BonusDamageModifier = effect.DamageModifier;
+                Target.BonusHealModifier = effect.HealModifier;
+                Target.BonusReflectChance += effect.ReflectChance;
+                Target.BonusReflectIntensity += effect.ReflectIntensity;
+                Target.BonusAc += effect.Str;
+                Target.OffensiveElementOverride = (Enums.Element)effect.OffensiveElement;
+                Target.DefensiveElementOverride = (Enums.Element)effect.OffensiveElement;
+
```

Comment:
null handling? on effect properties?

Reply:
Everything here is a value type and cannot be null and/or has defaults assigned in XML.

Reply:
OK